relay-metrics/src/aggregation.rs

```diff
@@ -1153,7 +1153,7 @@ impl Aggregator {
                             slf.merge_all(project_key, buckets).ok();
                         } else {
                             // NOTE: This means we drop buckets if the project state is expired.
-                            relay_log::trace!(
+                            relay_log::error!(
```

Comment:
This is technically not an additional metric, but I believe it's a useful change as we are currently debugging where metrics are dropped in the pipeline, and this is one of the known conditions where this could occur.

According to the statsd metric logged immediately below, we did not actually yet run into this condition, but at the same time if we did, we'd have no alerting for it. IMO Sentry errors are the right way to get alerted on error conditions, not building more statsd metrics and setting up explicit alerts for them.